WalletWasabi.Tests/UnitTests/Transactions/TransactionFactoryTests.cs

```diff
@@ -561,6 +561,20 @@ public void DoNotSignWatchOnly()
 		Assert.False(result.Signed);
 	}
 
+	[Fact]
+	public void ManyCoins()
+	{
+		var transactionFactory = ServiceFactory.CreateTransactionFactory(
+			DemoCoinSets.LotOfCoins);
+
+		using Key key = new();
+		var payment = new PaymentIntent(key, MoneyRequest.Create(Money.Coins(0.29943925m)));
+
+		Assert.Equal(ChangeStrategy.Auto, payment.ChangeStrategy);
+
+		Assert.Throws<InsufficientBalanceException>(() => transactionFactory.BuildTransaction(payment, new FeeRate(12m)));
```

Comment:
It is really not enough or it's just that it is impossible to pay that because the tx would be too big?

Reply:
Tx would be too big.